### PR TITLE
change baseclass for xbase formatter: simplify impls

### PR DESCRIPTION
**Pull request by cdietrich**
change baseclass for xbase formatter: simplify impls

### CODE REVIEW

File: org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java [hunk 1]
```diff
@@ -28,68 +28,63 @@ protected void _format(Entities entities, @Extension IFormattableDocument docume
...
-			document.append(document.prepend(operation.getType(), IHiddenRegionFormatter::noSpace),
-					IHiddenRegionFormatter::oneSpace);
+			document.append(regionFor(operation).keyword(")"), this::noSpace);
+			document.append(document.prepend(operation.getType(), this::noSpace), this::oneSpace);
 			document.format(operation.getType());
 		} else {
-			document.append(textRegionExtensions.regionFor(operation).keyword(")"), IHiddenRegionFormatter::oneSpace);
+			document.append(regionFor(operation).keyword(")"), this::oneSpace);
 		}
 		document.format(operation.getBody());
```
Comment (cdietrich) [inline org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java:89]: if you expand 5 lines you will see the if else dispatching i dislike and why i want to have the option to polymorphic dispatch here

Comment (szarnekow) [reply org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java:89]: I see - the concern is about the duality of xtend-compiled code and reflective polymorphic dispatching?

Comment (cdietrich) [reply org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java:89]: yes. unfortunately it is not possible to call Grandparent.super.doit(xxxx)

Comment (szarnekow) [reply org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java:89]: We could make a final protected method superDoIt() in the immediate subclass. Not pretty, though.

Comment (cdietrich) [reply org.eclipse.xtext.web.example.entities/src/org/eclipse/xtext/web/example/entities/formatting2/EntitiesFormatter.java:89]: that would require it to be copy&pasted.
this is why extracted the method in xtext-core

**Pull request merged**